theano/gof/cc.py

```diff
@@ -1445,22 +1464,116 @@ def compile_cmodule(self, location=None):
+                    f = open(os.path.join(location, 'makefile'), 'w')
+                    print(makefile, file=f)
+                    f.close()
+                    # Put the command line in the header code so that
+                    # other people know how to recompile the shared lib
+                    mod_exec.add_header_code(
+                        "//command line used to compile the shared lib: \n" +
+                        "//" + ' '.join(
+                            c_compiler.compile_command(
+                                module_name=mod.code_hash,
+                                location=location,
+                                include_dirs=self.header_dirs(),
+                                lib_dirs=self.lib_dirs(),
+                                libs=libs,
+                                preargs=preargs)[2]))
+
+                    # Make the executable link to the shared lib.
+                    preargs.append(os.path.join(location, mod.code_hash + "." +
+                                                cmodule.get_lib_extension()))
+                    # Make the executable
+                    mod_exec.add_header_code(
+                        "//command line used to compile the executable: \n" +
+                        "//" + ' '.join(
+                            c_compiler.compile_command(
+                                module_name=mod_exec.code_hash,
+                                location=location,
+                                include_dirs=self.header_dirs(),
+                                lib_dirs=self.lib_dirs(),
+                                libs=libs,
+                                preargs=preargs,
+                                shared=False, py_module=False,
+                                code_filename='exec.cpp',
+                                out_filename='exec')[2]))
+
+                    # compile the dynamic python module.
+                    src_code = mod_exec.code(executable=True)
+                    c_compiler.compile_str(
+                        module_name=mod_exec.code_hash,
+                        src_code=src_code,
+                        location=location,
+                        include_dirs=self.header_dirs(),
+                        lib_dirs=self.lib_dirs(),
+                        libs=libs,
+                        preargs=preargs,
+                        shared=False, py_module=False,
+                        code_filename='exec.cpp',
+                        out_filename='exec')
+                    mod_exec.gen_header(os.path.join(location, 'exec.h'))
+                    if sys.platform == "win32" and False:
+                        # I don't know why it work now, but this was needed in the past.
+                        # As this is complicated to find how to do it, I keep it here
+                        # just in case.
+                        mt = r"C:\Program Files (x86)\Microsoft SDKs\Windows\v7.0A\Bin\mt.exe"
```

Comment:
This is not ok since it will only work for people that have the Windows 7 SDK (32-bits).  We try to be a bit more flexible.